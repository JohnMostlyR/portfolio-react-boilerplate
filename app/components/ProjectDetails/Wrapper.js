import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
`;

const Wrapper = styled.div`
  opacity: ${({ animate }) => (animate ? 0 : 1)};
  animation-delay: 0ms;
  animation-direction: normal;
  animation-duration: 1s;
  animation-fill-mode: both;
  animation-name: ${({ animate }) => (animate ? fadeIn : 'none')};
  animation-timing-function: ease-in-out;

  @supports (display: grid) {
    display: grid;
    grid-area: mainContent;
    grid-template-areas:
      'pageHeader'
      'pageContent';
  }
`;

Wrapper.propTypes = {
  animate: PropTypes.bool,
};

Wrapper.defaultProps = {
  animate: true,
};

export default Wrapper;
