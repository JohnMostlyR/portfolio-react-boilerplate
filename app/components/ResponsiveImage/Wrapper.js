import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * The responsive image pattern automatically reserves the space
 * needed for an image to be inserted
 *
 *
 * 1. Setting the height to zero collapses the div so we don't
 *    need to absolutely position the image
 * 2. Stop images 1px difference between the reserved space
 *    and the actual image from showing
 */
const Wrapper = styled.div`
  position: relative;
  height: 0; /* [1] */
  overflow: hidden; /* [2] */
  background-color: ${props =>
    props.noPlaceholder ? 'transparent' : '#f0f0f0'};

  /**
   * 1. Force the image to sit on top of the placeholder
   * 2. Force the image to fill the available space
   *
   * The exact combination of [1] and [2] is needed to stop the
   * image from being 1px too short, the root cause of which is
   * down to the browser's rendering of sub-pixels
   */
  img {
    position: absolute; /* [1] */
    height: 100%; /* [2] */
    width: 100%; /* [2] */
  }

  /**
   * Aspect radio modifiers
   */
  padding-bottom: ${props => {
    switch (props.ratio) {
      case '1by1':
        return '100%';
      case '4by3':
        return '75%';
      case '16by9':
        return '56.25%';
      default:
        return 0;
    }
  }};
`;

Wrapper.propTypes = {
  noPlaceholder: PropTypes.bool,
  ratio: PropTypes.oneOf(['1by1', '4by3', '16by9']),
};

Wrapper.defaultProps = {
  noPlaceholder: false,
};

export default Wrapper;
