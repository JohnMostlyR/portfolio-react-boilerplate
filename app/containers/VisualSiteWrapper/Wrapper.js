import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100%;

  @supports (display: grid) {
    display: grid;
    grid-template-rows: auto 1fr;
    grid-template-areas:
      'header'
      'main';
  }
`;

export default Wrapper;
