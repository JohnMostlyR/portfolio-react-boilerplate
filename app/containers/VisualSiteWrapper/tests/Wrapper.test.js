import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Wrapper from '../Wrapper';

describe('<Wrapper />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<Wrapper />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
