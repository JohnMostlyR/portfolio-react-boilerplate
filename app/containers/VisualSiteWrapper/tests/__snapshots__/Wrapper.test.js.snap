// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render and match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-rows: auto 1fr;
    grid-template-areas: 'header' 'main';
  }
}

<div
  className="c0"
/>
`;
